custom/icds_reports/filters.py

```diff
@@ -37,38 +37,63 @@ def loc_to_json(loc, project):
 
     project = Domain.get_by_name(domain)
 
-    locations = SQLLocation.root_locations(domain)
+    if user.has_permission(domain, 'access_all_locations'):
+        accessible_root_locations = SQLLocation.root_locations(domain)
+    else:
+        accessible_root_locations = [(user.sql_location
+                                      .get_ancestors(include_self=True)
+                                      .get(location_type__code='state'))
+                                     ]
+
     if not show_test:
-        locations = [
-            loc for loc in locations if loc.metadata.get('is_test_location', 'real') != 'test'
-        ]
+        accessible_root_locations = [loc for loc in accessible_root_locations
+                                     if loc.metadata.get('is_test_location', 'real') != 'test']
```

Comment:
looks like a bug in the original implementation that filters out for test root locations, but does not do that again if any of the child location is marked as test for a non-test state. Not saying you should fix it here but does look like something that can go wrong.

Reply:
hey, I think this is doing similarly we are doing in the dashboard and might have spun off from the thinking of the system that no lower-level location is marked as a test if its state is nontest.
Although this is definitely just a functional level agreement which is definitely not restricted by any UI. 

I think I will just use this opportunity to fix this as well and make this check on children as well.